### PR TITLE
feat: adds workflow to trigger revalidation of docs pages

### PR DESCRIPTION
**Pull request by BRKalow**
Adds a workflow to trigger revalidation of changed docs pages. It does this by detecting changed files and sending the changed files list in a POST request to our `revalidate-docs` endpoint. If the `manifest.json` file is changed, it will send a request with `"paths": "*"`, which will trigger revalidation of every path.

Requires https://github.com/clerkinc/clerk-marketing/pull/715 to be merged into the assembly branch.

### CODE REVIEW

File: .github/workflows/revalidate.yml [hunk 1]
```diff
@@ -37,4 +37,4 @@ jobs:
           curl -X POST ${{ env.CLERK_COM_BASE_URL }}/api/revalidate-docs \
             -H "Content-Type: application/json" \
             -H "Authorization: Bearer ${{ secrets.REVALIDATE_DOCS_SECRET }}" \
-            -d '{"paths": ${{ steps.paths-to-revalidate.outputs.paths }}}'
+            -d "{\"paths\": ${{ steps.paths-to-revalidate.outputs.paths }}}""
```
Comment (alexisintech) [inline .github/workflows/revalidate.yml:40]: two closing quotes?

Comment (BRKalow) [reply .github/workflows/revalidate.yml:40]: That aint right! Good catch 🙏

File: .github/workflows/revalidate.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        run: |
+          paths_to_revalidate=$(echo ${{ contains(steps.changed-files.outputs.all_changed_files, 'docs/manifest.json') && '\\"*\\"' || toJSON(steps.changed-files.outputs.all_changed_files) }} | sed 's/.mdx//g' | sed 's/\/index//g')
+          echo "paths=$paths_to_revalidate" >> "$GITHUB_OUTPUT"
+
+      - name: Trigger revalidate
+        run: |
+          curl -X POST ${{ env.CLERK_COM_BASE_URL }}/api/revalidate-docs \
+            -H "Content-Type: application/json" \
+            -H "Authorization: Bearer ${{ secrets.REVALIDATE_DOCS_SECRET }}" \
+            -d "{\"paths\": ${{ steps.paths-to-revalidate.outputs.paths }}}""
```
Comment (BRKalow) [inline .github/workflows/revalidate.yml:40]:
```suggestion
            -d "{\"paths\": ${{ steps.paths-to-revalidate.outputs.paths }}}"
```

**Review by alexisintech — APPROVED** [review]
😸💖

**Pull request merged**